materialize-sql/table_mapping.go

```diff
@@ -137,6 +142,31 @@ func ResolveTable(shape TableShape, dialect Dialect) (Table, error) {
 	for index, col := range table.Columns() {
 		var err error
 
+		// TODO(whb): This bit of hackery is to provide backwards compatibility for materializations
+		// that use numeric-as-string fields with numeric formats which were created before support
+		// for materializing these fields as numeric values was added. It provides an escape hatch
+		// via the field config to ignore the string formatting and continue materializing the
+		// fields as a regular string, which will allow such materializations to continue working.
+		// It is not meant to be user-facing and is configured as-needed by Estuary support staff.
+
+		// We should remove this when there are no longer any materializations that need it (there's
+		// only 1 known case where it is needed currently), or a more comprehensive backwards
+		// compatibility layer is added to the materialization dialects.
+		if col.RawFieldConfig != nil {
+			var options fieldConfigOptions
+			if err := json.Unmarshal(col.RawFieldConfig, &options); err != nil {
+				return Table{}, fmt.Errorf("unmarshaling RawFieldConfig for column %s of %s: %w", col.Field, shape.Path, err)
+			} else if options.IgnoreStringFormat {
+				log.WithFields(log.Fields{
+					"collection": table.Source.String(),
+					"field":      col.Field,
+					"format":     col.Projection.Inference.String_.Format,
+				}).Info("ignoring string format for field")
+
+				col.Projection.Inference.String_.Format = ""
```

Comment:
By my reading it's possible for `col.Projection.Inference.String_` to be nil, at least according to the type system. And if that happens when `IgnoreStringFormat` is set this code will panic. That seems like a reasonable response IMO to that corner case, but I wanted to make sure it's intentional.

Reply:
You are right that `col.Projection.Inference.String_` can be `nil` if the types don't include `string`, and that would result in a panic if the field config is set in this way. That kind of failure wouldn't be very user-friendly, but considering the expected usage of this option I'm not too worried about it.